src/main/java/org/pih/petl/api/JobExecutor.java

```diff
@@ -20,27 +24,54 @@
 
     private static Log log = LogFactory.getLog(JobExecutor.class);
 
-    private final Integer maxConcurrentJobs;
+    private final EtlService etlService;
     private final ScheduledExecutorService executorService;
 
-    public JobExecutor(Integer maxConcurrentJobs) {
-        this.maxConcurrentJobs = maxConcurrentJobs;
+    public JobExecutor(EtlService etlService, Integer maxConcurrentJobs) {
+        this.etlService = etlService;
         this.executorService = Executors.newScheduledThreadPool(maxConcurrentJobs);
     }
 
-    /**
-     * Execute a single task
-     */
-    public void execute(JobExecutionTask task) throws InterruptedException, ExecutionException {
-        executeInSeries(Collections.singletonList(task));
-    }
-
     public void shutdown() {
         if (!executorService.isShutdown()) {
             executorService.shutdownNow();
         }
     }
 
+    /**
```

Comment:
All of the execution logic is moved into the JobExecutor.  This executeJob method was moved here from etlservice and adapted with additional tracking of statuses.